### PR TITLE
ipatest:Test if getcert creates cacert file with -F option

### PR DESCRIPTION
**Pull request by mrizwan93**
It took longer to create the cacert file in older version.
restarting the certmonger service creates the file at the location
specified by -F option. This fix is to check that cacert file
creates immediately after certificate goes into MONITORING state.

related: https://pagure.io/freeipa/issue/8105

Signed-off-by: Mohammad Rizwan Yusuf <myusuf@redhat.com>

### CODE REVIEW

File: ipatests/test_integration/test_installation.py [hunk 1]
```diff
@@ -1208,3 +1208,36 @@ def test_replica_install_against_server(self):
...
+        cmd_arg = ['ipa-getcert request',
+                   '-f', '/etc/pki/tls/certs/test.pem',
+                   '-k', '/etc/pki/tls/private/test.key',
+                   '-K', 'test/`hostname`', '-F',
+                   '/etc/pki/tls/test.CA']
+        result = self.clients[0].run_command([cmd_arg])
+        request_id = re.findall(r'\d+', result.stdout_text)
+
+        # check if certificate is in MONITORING state
+        status = wait_for_request(self.master, request_id[0], 50)
```
Comment (rcritten) [inline ipatests/test_integration/test_installation.py]: self.master -> self.clients[0]

File: ipatests/test_integration/test_installation.py [hunk 2]
```diff
@@ -1208,3 +1208,36 @@ def test_replica_install_against_server(self):
...
+        """Test if getcert creates cacert file with -F option
+
+        It took longer to create the cacert file in older version.
+        restarting the certmonger service creates the file at the location
+        specified by -F option. This fix is to check that cacert file
+        creates immediately after certificate goes into MONITORING state.
+
+        related: https://pagure.io/freeipa/issue/8105
+        """
+        cmd_arg = ['ipa-getcert request',
```
Comment (rcritten) [inline ipatests/test_integration/test_installation.py]: 'ipa-getcert', 'request',

File: ipatests/test_integration/test_installation.py [hunk 3]
```diff
@@ -1208,3 +1208,36 @@ def test_replica_install_against_server(self):
                                             self.replicas[0].hostname],
                                            stdin_text=dirman_password)
         assert self.replicas[0].hostname not in cmd.stdout_text
+
+
+class TestInstallMasterClient(IntegrationTest):
```
Comment (kaleemsiddiqu) [inline ipatests/test_integration/test_installation.py]: This scenario is not related with installation, so IMO should not be part of test_installation.py. We can add a new test file test_cert.py which can contain cert related tests.

Comment (mrizwan93) [reply ipatests/test_integration/test_installation.py]: done

File: ipatests/test_integration/test_cert.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        restarting the certmonger service creates the file at the location
+        specified by -F option. This fix is to check that cacert file
+        creates immediately after certificate goes into MONITORING state.
+
+        related: https://pagure.io/freeipa/issue/8105
+        """
+        cmd_arg = ['ipa-getcert', 'request',
+                   '-f', '/etc/pki/tls/certs/test.pem',
+                   '-k', '/etc/pki/tls/private/test.key',
+                   '-K', 'test/`hostname`', '-F',
```
Comment (rcritten) [inline ipatests/test_integration/test_cert.py]: You'll need to set the hostname using self.clients[0].hostname

**Comment by rcritten** [comment]
I'd prefer it if you completely remove the test_installation test in the first patch so it doesn't have to be removed in the unrelated "move wait_for_requests" patch.

**Comment by rcritten** [comment]
Looks good, thanks. Please remove the temp commit and I'll ack.
